sdks/go/pkg/beam/testing/teststream/teststream_test.go

```diff
@@ -0,0 +1,116 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package teststream
+
+import (
+	"bytes"
+	"reflect"
+	"testing"
+
+	"github.com/apache/beam/sdks/go/pkg/beam"
+)
+
+func TestNewConfig(t *testing.T) {
+	con := NewConfig()
+	if con.elmType != nil {
+		t.Errorf("type is not correct, expected nil, got %v", con.elmType)
+	}
+	if len(con.events) != 0 {
+		t.Errorf("config has too many elements, expected 0, got %v", len(con.events))
+	}
+	if con.endpoint.Url != "" {
+		t.Errorf("config has URL endpoint when it should be empty")
+	}
+}
+
+func TestAdvanceWatermark(t *testing.T) {
+	con := NewConfig()
+	con.AdvanceWatermark(500)
+	if w := con.watermark; w != 500 {
+		t.Errorf("default watermark expected 500, got %v", w)
+	}
+	if len(con.events) != 1 {
+		t.Fatalf("expected only 1 event in config, got %v", len(con.events))
+	}
+	if eventWatermark := con.events[0].GetWatermarkEvent().NewWatermark; eventWatermark != 500 {
+		t.Errorf("expected watermark in event was 500, got %v", eventWatermark)
```

Comment:
Style nit: Go idiom prefers using "want" over "expected". Not required if want is more convoluted to use of course.

Reply:
Done.